src/main/java/org/broadinstitute/hellbender/utils/nio/NioFileCopier.java

```diff
@@ -0,0 +1,624 @@
+     */
+    public boolean isOverwriteExisting() {
+        return overwriteExisting;
+    }
+
+    /**
+     * Sets whether this {@link NioFileCopier} will overwrite {@link #dest} when copying.
+     */
+    public NioFileCopier setOverwriteExisting(final boolean overwriteExisting) {
+        this.overwriteExisting = overwriteExisting;
+
+        return this;
+    }
+
+    /**
+     * @return {@code true} iff this {@link NioFileCopier} will log intermediate progress when copying.  {@code false} otherwise.
+     */
+    public boolean isDoLogProgress() {
+        return doLogProgress;
+    }
+
+    /**
+     * Sets whether this {@link NioFileCopier} will log intermediate progress when copying.
+     */
+    public NioFileCopier setDoLogProgress(final boolean doLogProgress) {
+        this.doLogProgress = doLogProgress;
+
+        return this;
+    }
+
+    /**
+     * {@code true} if this {@link NioFileCopier} will verbosely log intermediate progress when copying.
+     * This is independent of whether intermediate progress logging is enabled overall.
+     * Intermediate progress logging is mediated by {@link #doLogProgress}.
+     * @return {@code true} iff this {@link NioFileCopier} will verbosely log intermediate progress when copying.  {@code false} otherwise.
+     */
+    public boolean isLogVerbose() {
+        return logVerbose;
+    }
+
+    /**
+     * Sets whether this {@link NioFileCopier} will verbosely log intermediate progress when copying.
+     * This is independent of whether intermediate progress logging is enabled overall.
+     * Intermediate progress logging is mediated by {@link #doLogProgress}.
+     */
+    public NioFileCopier setLogVerbose(final boolean logVerbose) {
+        this.logVerbose = logVerbose;
+
+        return this;
+    }
+
+    /**
+     * @return {@code true} iff this {@link NioFileCopier} will copy completely silently, with no logging output whatsoever.  {@code false} otherwise.
+     */
+    public boolean isSilentCopy() {
+        return silentCopy;
+    }
+
+    /**
+     * Sets whether this {@link NioFileCopier} will copy completely silently, with no logging output whatsoever.
+     */
+    public NioFileCopier setSilentCopy(final boolean silentCopy) {
+        this.silentCopy = silentCopy;
+
+        return this;
+    }
+
+    /**
+     * Get the last checksum calculated against {@link #dest}.
+     * If no checksum has been calculated, will return an empty {@link String}.
+     * @return The last checksum calculated against {@link #dest}.
+     */
+    public String getLatestChecksum() {
+        return latestChecksum;
+    }
+
+    /**
+     * Get the last {@link ChecksumCalculator} used to calculate the checksum of {@link #dest}.
+     * If no checksum has been calculated, will return {@code null}.
+     * @return The last {@link ChecksumCalculator} used to calculate the checksum of {@link #dest}.
+     */
+    public ChecksumCalculator getLatestChecksumCalculator() {
```

Comment:
The "latest" part of this seems strange to me.  Maybe it's better to expose a query by digest so you can retrieve multiple checksums.  Or just drop to "getChecksum" if multiples aren't supported.

Reply:
Yeah - I pulled out the latest checksum business.

Now only 1 checksum is stored.